pulp_deb/app/serializers/content_serializers.py

```diff
@@ -674,3 +704,410 @@ class PackageReleaseComponentSerializer(NoArtifactContentSerializer):
+        """
+        model = self.Meta.model(**source_data)
+        if "relative_path" not in data:
+            data["relative_path"] = model.derived_path(model.derived_dsc_filename())
+        elif not os.path.basename(data["relative_path"]) == model.derived_dsc_filename():
+            raise ValidationError(
+                _("Invalid relative_path provided '{}', filename '{}' do not match.").format(
+                    data["relative_path"], model.derived_dsc_filename()
+                )
+            )
+
+        content = SourcePackage.objects.filter(source=data["source"], version=data["version"])
+        if content.exists():
+            raise ValidationError(
+                _(
+                    "There is already a DSC file with version '{version}' and source name "
+                    "'{source}'."
+                ).format(version=data["version"], source=data["source"])
+            )
+
+        artifacts = {data["relative_path"]: data["artifact"]}
+        for source in data["checksums_sha256"]:
+            content = Artifact.objects.filter(sha256=source["sha256"], size=source["size"])
+            if not content.exists():
+                raise ValidationError(
+                    _(
+                        "A source file is listed in the DSC file but is not yet available '{name}'"
+                        " and sha256 '{sha256}'."
+                    ).format(name=source["name"], sha256=source["sha256"])
+                )
+            artifacts[
+                os.path.join(os.path.dirname(data["relative_path"]), source["name"])
+            ] = content.first()
+
+        data["artifacts"] = artifacts
+        return data
+
+    class Meta:
+        fields = MultipleArtifactContentSerializer.Meta.fields + (
+            "artifact",
+            "relative_path",
+            "format",
+            "source",
+            "binary",
+            "architecture",
+            "version",
+            "maintainer",
+            "uploaders",
+            "homepage",
+            "vcs_browser",
+            "vcs_arch",
+            "vcs_bzr",
+            "vcs_cvs",
+            "vcs_darcs",
+            "vcs_git",
+            "vcs_hg",
+            "vcs_mtn",
+            "vcs_snv",
+            "testsuite",
+            "dgit",
+            "standards_version",
+            "build_depends",
+            "build_depends_indep",
+            "build_depends_arch",
+            "build_conflicts",
+            "build_conflicts_indep",
+            "build_conflicts_arch",
+            "package_list",
+        )
+        model = SourcePackage
+
+
+class SourcePackageReleaseComponentSerializer(NoArtifactContentSerializer):
+    """
+    A Serializer for SourcePackageReleaseComponent.
+    """
+
+    source_package = DetailRelatedField(
+        help_text="Source package that is contained in release_comonent.",
+        many=False,
+        queryset=ReleaseComponent.objects.all(),
```

Comment:
SourcePackages?

Reply:
I do see a typo above "release_comonent", but I am not understanding your comment, maybe you can expand some. NOTE SourcePackage is a Debian terminology (https://wiki.debian.org/Packaging/SourcePackage), am I missing a pluralization or something else?

Reply:
This line should be:

```suggestion
        queryset=SourcePackage.objects.all(),
```

Reply:
Sorry for the confusingly short comment. Thanks David, for the clarification.